### ISSUE TITLE
Redis storage breaks with aioredis 1.0

### ISSUE THREAD

**Issue by hubo1016** [aio-libs/aiohttp-session#231]
```
2017-11-20 03:35:36,263 ERROR    aiohttp.server Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 416, in start
    resp = yield from self._request_handler(request)
  File "/usr/local/lib/python3.6/site-packages/aiohttp/web.py", line 323, in _handle
    resp = yield from handler(request)
  File "/usr/local/lib/python3.6/site-packages/aiohttp_session/__init__.py", line 142, in factory
    yield from storage.save_session(request, response, session)
  File "/usr/local/lib/python3.6/site-packages/aiohttp_session/redis_storage.py", line 60, in save_session
    with (yield from self._redis) as conn:
TypeError: 'ConnectionsPool' object is not iterable
```

There are breaking changes in aioredis 1.0, especially Python 3.3, 3.4 supports are dropped, See https://github.com/aio-libs/aioredis/releases/tag/v1.0.0

May be fixed by limiting the aioredis version or upgrading to the new API?

**Comment by panagiks** [comment]
There's already talk in #218 for upgrading to the new aioredis API. So I think it would solve the problem (but not sure what the best sort-term solution would be).

@asvetlov I'll have a look into porting to the new API it in the next couple days. Also since I see that a couple aio-libs go towards dropping support for Python3.4 should aiohttp-session also migrate to the new async/await syntax?

**Comment by asvetlov** [comment]
I've published aiohttp-session 1.2.1 with aioredis requirement `<1.0`.

Let's port a new redis API, get rid of `yield from` and make a major release.
